sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -771,13 +772,26 @@ def _build_setup_package(setup_file,  # type: str
     try:
       os.chdir(os.path.dirname(setup_file))
       if build_setup_args is None:
-        build_setup_args = [
-            Stager._get_python_executable(),
-            os.path.basename(setup_file),
-            'sdist',
-            '--dist-dir',
-            temp_dir
-        ]
+        # if build is installed in the user env, use it to
+        # build the sdist else fallback to legacy setup.py sdist call.
+        if importlib.util.find_spec('build'):
```

Comment:
Are you aware of any cases where the new process might fail to build but old one would succeed? I suppose users would have a workaround to uninstall `build`.

Reply:
The new process calls the old process under the hood in a way.